### PR TITLE
xray16: new port

### PR DESCRIPTION
**Pull request by Martinfx**

### CODE REVIEW

File: xray-16/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+PORTNAME=       xray-16
+DISTVERSION=    2188-november-2023-rc1
+CATEGORIES=     games
+
+MAINTAINER=     freebsd@sysctl.cz
+COMMENT=        game engine for stalker games
```
Comment (Xottab-DUTY) [inline xray-16/Makefile:6]: Please, use the official project description.

File: xray-16/Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+LICENSE=        UNKNOWN
+LICENSE_NAME=   unknown
+LICENSE_FILE=   ${WRKSRC}/License.txt
+LICENSE_PERMS=  dist-mirror no-dist-sell pkg-mirror no-pkg-sell auto-accept
+
+USES=           cmake localbase lua:51
```
Comment (Xottab-DUTY) [inline xray-16/Makefile:13]: OpenXRay is not compatible with Lua5.1, we use custom LuaJIT.

File: xray-16/files/patch-Externals_renderdoc_renderdoc__app.h
Comment (Xottab-DUTY) [inline xray-16/files/patch-Externals_renderdoc_renderdoc__app.h:1]: Also, I think that RenderDoc doesn't support FreeBSD, right?
It's better to #ifdef out the inclusion of 	renderdoc_app.h	 in our renderer code then.
In fact, we did exactly that in the latest commits, though, unfortunately a new tag is not ready yet.

File: xray-16/Makefile (hunk 1 above)
Comment (Martinfx) [reply xray-16/Makefile]: I added description

File: xray-16/Makefile (hunk 2 above)
Comment (Martinfx) [reply xray-16/Makefile]: I have problem with linking -lLua51
FAILED: /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrScriptEngine.so 
: && /usr/local/bin/clang++18 -fPIC -pipe -g -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -fdelayed-template-parsing -Wno-unused-command-line-argument -Wno-inconsistent-missing-override -stdlib=libc++ -Wno-attributes -Wl,--no-undefined   -mfpmath=sse -msse3 -Og -Wall -Wextra -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-parentheses -Wno-unused-label -Wno-unused-parameter -Wno-switch -pipe -g -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include  -fstack-protector-strong -fuse-ld=lld -shared -Wl,-soname,xrScriptEngine.so -o /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrScriptEngine.so src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_callStack.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_debugger.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_debugger_threads.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_lua_helper.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/BindingsDumper.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_engine.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/ScriptEngineScript.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/ScriptExporter.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/pch.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/xrScriptEngine.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/LuaStudio/LuaStudio.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_process.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_thread.cpp.o src/xrScriptEngine/CMakeFiles/xrScriptEngine.dir/script_stack_tracker.cpp.o -L/usr/local/lib -Wl,-rpath,"\$ORIGIN:/usr/local/lib:/usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug:"  /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrAPI.so  /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrLuaFix.a  /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrLuabind.a  /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrLuajit-debug.so  /usr/home/maxfx/Documents/FreeBSD-Ports/xray-16/work/xray-16-2188-november-2023-rc1/bin/x64/Debug/xrCore.so  -lpthread  -lexecinfo  /usr/local/lib/libSDL2.so  -lLua51  -Wl,-rpath-link,/usr/local/lib && :
ld.lld: error: unable to find library -lLua51
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
*** Error code 1

File: xray-16/files/patch-Externals_renderdoc_renderdoc__app.h
Comment (Martinfx) [reply xray-16/files/patch-Externals_renderdoc_renderdoc__app.h:1]: Yes, Renderdoc dosent support FreeBSD

Comment (Xottab-DUTY) [inline xray-16/files/patch-Externals_renderdoc_renderdoc__app.h:1]: Since you are using latest commit, this patch should not be needed

**Pull request merged**